engine/class_modules/sc_warrior.cpp

```diff
@@ -4313,6 +4324,24 @@ struct ignore_pain_t : public warrior_spell_t
   }
 };
 
+// A free Ignore Pain from Bastion of Might when Avatar is cast
+struct ignore_pain_bom_t : public ignore_pain_t
```

Comment:
Did you check how it interacts with spells that interact with normal Ignore Pains like bloodsport, vengeance, or never surrender? If it is different from a regular Ignore Pain, you will need to override execute too.

Reply:
No, not extensively, will try and find time